### PR TITLE
All finalized entities

### PR DESCRIPTION
**Pull request by Eshchock1**
### Summary
This PR includes the final versions of all entities excluding GameState. 

*These entities were made by everyone on the team*

Entities added: 
- Maze 
- DesignableMaze
- PublishedMaze
- User
- Player
- Designer
- UserFactory

### CODE REVIEW

**Review by jchang0112 — DISMISSED** [review]

File: src/main/java/entities/DesignableMaze.java [hunk 1]
```diff
@@ -12,4 +16,65 @@ public void placeWall(int row, int col) {
...
+            }
+        }
+        this.state[row][col] = Maze.ENCODING.get("start");
+        startLocation[0] = row;
+        startLocation[1] = col;
+    }
+    public void placeEnd(int row, int col) {
+        for (int i = 0; i < this.getNumRow(); i++) {
+            for (int j = 0; j < this.getNumCol(); j++) {
+                if (this.state[i][j] == 'E'){
```
Comment (edweenie123) [inline src/main/java/entities/DesignableMaze.java]: Reference the encoding instead of writing 'E" directly

**Review by daexs — APPROVED** [review]

File: src/main/java/entities/DesignableMaze.java (hunk 1 above)
Comment (Eshchock1) [reply src/main/java/entities/DesignableMaze.java]: Fixed along with line 23.

File: src/main/java/entities/DesignableMaze.java [hunk 2]
```diff
@@ -12,4 +16,65 @@ public void placeWall(int row, int col) {
     public void deleteWall(int row, int col) {
         this.state[row][col] = Maze.ENCODING.get("empty");
     }
-}
+
+    public void placeStart(int row, int col) {
+        for (int i = 0; i < this.getNumRow(); i++) {
+            for (int j = 0; j < this.getNumCol(); j++) {
+                if (this.state[i][j] == 'S'){
```
Comment (edweenie123) [inline src/main/java/entities/DesignableMaze.java]: reference the encoding instead of writing 'S'

Comment (Eshchock1) [reply src/main/java/entities/DesignableMaze.java]: Fixed

**Review by oscartuvey — APPROVED** [review]

**Review by edweenie123 — APPROVED** [review]

**Pull request merged**